cmd/os-instrumented.go

```diff
@@ -105,7 +105,7 @@ func osTrace(s osMetric, startTime time.Time, duration time.Duration, path strin
 
 func updateOSMetrics(s osMetric, paths ...string) func(err error) {
 	if globalTrace.NumSubscribers(madmin.TraceOS) == 0 {
-		return func(err error) { globalOSMetrics.time(s) }
+		return func(err error) { globalOSMetrics.time(s)() }
```

Comment:
Shouldn't the function `osMetrics.time` return a `func(err error)` ?

Reply:
Yes, if I am not mistaken, `return func(err error) { globalOSMetrics.time(s)() }` can be replaced with `return globalOSMetrics.time(s)`

I don't see why not